src/services/expenses.services.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+const { getUserById } = require('./user.services');
+
+let expenses = [];
+
+const getExpenses = (userId, categories = [], from, to) => {
+  let filteredExpenses = [...expenses];
+
+  if (userId) {
+    const id = +userId;
+
+    filteredExpenses = filteredExpenses.filter(
+      (expense) => expense.userId === id,
+    );
+  }
+
+  if (categories.length > 0) {
+    const categoriesList = categories.split(',');
+
+    filteredExpenses = filteredExpenses.filter((expense) => {
+      return categoriesList.includes(expense.category);
+    });
+  }
+
+  if (from) {
+    filteredExpenses = filteredExpenses.filter(
+      (expense) => new Date(expense.spentAt) >= new Date(from),
+    );
+  }
+
+  if (to) {
+    filteredExpenses = filteredExpenses.filter(
+      (expense) => new Date(expense.spentAt) <= new Date(to),
+    );
+  }
```

Comment:
You use filter at least three times here. is it possible to combine it to iterate only once?